### PR TITLE
chore(RHTAPWATCH-150): go source checks

### PR DESCRIPTION
**Pull request by ShakedAviv50313**
add go source checks for the repo

### CODE REVIEW

**Comment by ifireball** [comment]
This is probably based on a very old version of the code - you can see the pre-commit action is not even running, please rebase on top of the latest code.

**Comment by ifireball** [comment]
Did you verify that the checks can actually catch issues?
Are we sure all our go files are checked?

File: scripts/runner.go [hunk 1]
```diff
@@ -46,3 +46,5 @@ func GetRepoRootDir() (string, error) {
 	dirPath := filepath.Dir(filepath.Dir(filename))
 	return dirPath, nil
 }
+
+
```
Comment (Omeramsc) [inline scripts/runner.go:50]: are those a result of changes required by the go source checks?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -15,3 +15,9 @@ repos:
         stages: [commit-msg]
     repo: https://github.com/jorisroovers/gitlint
     rev: v0.18.0
+  - hooks:
+      - id: go-fmt-import
+      - id: go-vet
+      - id: go-lint
+    repo: https://github.com/Bahjat/pre-commit-golang
+    rev: v1.0.2
```
Comment (Omeramsc) [inline .pre-commit-config.yaml:23]: 1.0.3 is available, any reason not to use it?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -15,3 +15,9 @@ repos:
         stages: [commit-msg]
     repo: https://github.com/jorisroovers/gitlint
     rev: v0.18.0
+  - hooks:
```
Comment (Omeramsc) [inline .pre-commit-config.yaml:18]: seems like this repo has also `go-unit-tests` and other hooks, worth looking if we can get some value from them

File: scripts/runner.go (hunk 1 above)
Comment (ShakedAviv50313) [reply scripts/runner.go:50]: I'm trying to verify that the go checks in this PR are actually working in the ci, these lines will be removed once I'm sure the checks work

File: .pre-commit-config.yaml (hunk 3 above)
Comment (ifireball) [reply .pre-commit-config.yaml:18]: Given that our unit tests also try to spin up containers, I wouldn't try to scope that into this change

File: .pre-commit-config.yaml (hunk 2 above)
Comment (ShakedAviv50313) [reply .pre-commit-config.yaml]: It is no longer being used, now we're using golangci-lint and using it's latest version.

**Comment by yftacherzog** [comment]
@ShakedAviv50313 one of the files that you modified here was also modified since yesterday. You'd need to rebase.

It might be tricky.

**Review by yftacherzog — CHANGES_REQUESTED** [review]
@ShakedAviv50313 you mixed up the content of the commits when you rebased. Ping me and we sort it out.

**Review by Omeramsc — APPROVED** [review]
LGTM

**Review by ifireball — APPROVED** [review]

**Pull request merged**